plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystemMinio.java

```diff
@@ -0,0 +1,143 @@
+ */
+package io.trino.plugin.hive.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.containers.Minio;
+import io.trino.testing.minio.MinioClient;
+import io.trino.util.AutoCloseableCloser;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+
+import static io.trino.hadoop.ConfigurationInstantiator.newEmptyConfiguration;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestTrinoS3FileSystemMinio
+        extends AbstractTestTrinoS3FileSystemObjectStorage
+{
+    public static final String MINIO_ACCESS_KEY = "accesskey";
+    public static final String MINIO_SECRET_KEY = "secretkey";
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+    private String bucketName = "trino-ci-test";
+
+    private Minio minio;
+
+    private MinioClient minioClient;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        minio = Minio.builder()
+                .withEnvVars(ImmutableMap.<String, String>builder()
+                        .put("MINIO_ACCESS_KEY", MINIO_ACCESS_KEY)
+                        .put("MINIO_SECRET_KEY", MINIO_SECRET_KEY)
+                        .buildOrThrow())
+                .build();
+        closer.register(minio);
+        minio.start();
+
+        minioClient = closer.register(minio.createMinioClient());
+        minio.createBucket(bucketName);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        closer.close();
+    }
+
+    @Override
+    protected String getBucketName()
+    {
+        return bucketName;
+    }
+
+    @Override
+    protected Configuration defaultS3Configuration()
+    {
+        Configuration config = newEmptyConfiguration();
+        config.set("trino.s3.endpoint", minio.getMinioAddress());
+        config.set("trino.s3.access-key", MINIO_ACCESS_KEY);
+        config.set("trino.s3.secret-key", MINIO_SECRET_KEY);
+        config.set("trino.s3.path-style-access", "true");
+
+        return config;
+    }
+
+    @Test
+    public void testDeleteNonRecursivelyBucketRoot()
+            throws Exception
+    {
+        String testBucketName = "trino-delete-bucket-root";
+        minioClient.makeBucket(testBucketName);
+        String testBucketPath = "s3://" + testBucketName;
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI(testBucketPath), defaultS3Configuration());
+
+            AmazonS3 s3 = fs.getS3Client();
+            String filename1 = "file1.txt";
+            fs.createNewFile(new Path("s3://%s/%s".formatted(testBucketName, filename1)));
```

Comment:
nit: I would prefer to use a string literal when the variable is not going to use more than once.